client/src/utils/table.ts

```diff
@@ -89,3 +90,87 @@ export const predictGrades = (
   }
   return result;
 };
+
+export const predictedGradesErrorCheck = (
+  studentPredictedGrades: {[k: string]: {finalGrade: number}},
+  courseScale: GradingScale
+): RowError[] => {
+  return Object.entries(studentPredictedGrades).reduce(
+    (errorsArray, [modelId, grade]) => {
+      // if grade is a float
+      if (grade.finalGrade % 1 !== 0) {
+        errorsArray.push({
+          message: 'The predicted grade is not an integer',
+          type: 'InvalidPredictedGrade',
+          info: {
+            columnId: 'predictedFinalGrades',
+            modelId: modelId,
+          },
+        });
+      }
+      // if grade is out of range
+      if (
+        (courseScale === GradingScale.Numerical &&
+          !(grade.finalGrade >= 0 && grade.finalGrade <= 5)) ||
+        (courseScale === GradingScale.PassFail &&
+          !(grade.finalGrade >= 0 && grade.finalGrade <= 1)) ||
+        (courseScale === GradingScale.SecondNationalLanguage &&
+          !(grade.finalGrade >= 0 && grade.finalGrade <= 2))
+      ) {
+        errorsArray.push({
+          message: 'The predicted grade is out of range',
+          type: 'OutOfRangePredictedGrade',
+          info: {
+            columnId: 'predictedFinalGrades',
+            modelId: modelId,
+          },
+        });
+      }
+      return errorsArray;
+    },
+    [] as RowError[]
+  );
+};
+
+/**
+ * Calculates the total count of errors in the given row model based on the
+ * selected grading model.
+ *
+ * @param rowModel - The array of grouped student rows.
+ * @param selectedGradingModel - The selected grading model. Can be 'any' or a
+ *   number.
+ * @returns The total count of errors.
+ */
+export const getErrorCount = (
+  rowModel: GroupedStudentRow[],
+  selectedGradingModel: 'any' | number
+): number => {
+  let totalErrors = 0;
+
+  for (const row of rowModel) {
+    let rowErrorCount = 0;
```

Comment:
The `rowErrorCount` variable is unnecessary, just add to the `totalErrors` variable